transport/graphsync/graphsync.go

```diff
@@ -278,129 +207,64 @@ func (t *Transport) executeGsRequest(
 
 	// Used by the tests to listen for when a request completes
 	if t.completedRequestListener != nil {
-		t.completedRequestListener(channelID)
+		t.completedRequestListener(ch.channelID)
 	}
 
-	err := t.events.OnChannelCompleted(channelID, completeErr)
+	err := t.events.OnChannelCompleted(ch.channelID, completeErr)
 	if err != nil {
-		log.Error(err)
-	}
-}
-
-func (t *Transport) gsKeyFromChannelID(ctx context.Context, chid datatransfer.ChannelID) (graphsyncKey, error) {
-	for {
-		t.dataLock.RLock()
-		gsKey, ok := t.channelIDMap[chid]
-		if ok {
-			t.dataLock.RUnlock()
-			return gsKey, nil
-		}
-		pending, hasPending := t.pending[chid]
-		t.dataLock.RUnlock()
-		if !hasPending {
-			return graphsyncKey{}, datatransfer.ErrChannelNotFound
-		}
-		select {
-		case <-ctx.Done():
-			return graphsyncKey{}, datatransfer.ErrChannelNotFound
-		case <-pending:
-		}
```

Comment:
Refactored the code so that we no longer need the concept of "pending" - a channel is locked until it is ready